### ISSUE TITLE
Unit testing "Launcher" section needs update for JUnit 5

### ISSUE THREAD

**Issue by itzg** [etcd-io/jetcd#742]
The "Launcher" section in the README:

https://github.com/etcd-io/jetcd#launcher

seems to be out of date since now an [`EtcdClusterFactory`](https://github.com/etcd-io/jetcd/blob/jetcd-0.5.3/jetcd-launcher/jetcd-launcher/src/main/java/io/etcd/jetcd/launcher/EtcdClusterFactory.java) is used to create an `EtcdCluster`.

I believe the example snippet should be the following and I would be glad to submit a PR if this all sounds right:

```
@Rule public final EtcdClusterResource etcd = EtcdClusterFactory.buildCluster("test-etcd", 1, false);
Client client = Client.builder().endpoints(etcd.getClientEndpoints()).build();
```

**Comment by itzg** [comment]
I spoke too soon with replacement example. I am now finding that construct doesn't work as a junit rule:

```
org.junit.internal.runners.rules.ValidationError: The @ClassRule 'etcd' must implement TestRule.
```

**Comment by itzg** [comment]
I ended up needing to implement my own TestRule class, so it feels like I'm missing something:

```java
import io.etcd.jetcd.launcher.EtcdCluster;
import io.etcd.jetcd.launcher.EtcdClusterFactory;
import java.net.URI;
import java.util.List;
import org.junit.rules.TestRule;
import org.junit.runner.Description;
import org.junit.runners.model.Statement;

public class EtcdTestCluster implements TestRule {

  private final String clusterName;
  private final int nodes;
  private final boolean ssl;
  private EtcdCluster cluster;

  public EtcdTestCluster(String clusterName, int nodes) {
    this(clusterName, nodes, false);
  }

  public EtcdTestCluster(String clusterName, int nodes, boolean ssl) {
    this.clusterName = clusterName;
    this.nodes = nodes;
    this.ssl = ssl;
  }

  @Override
  public Statement apply(Statement base, Description description) {
    return new Statement() {
      @Override
      public void evaluate() throws Throwable {
        cluster = EtcdClusterFactory.buildCluster(clusterName, nodes, ssl);

        cluster.start();
        try {
          base.evaluate();
        } finally {
          cluster.close();
          cluster = null;
        }
      }
    };
  }

  public List<URI> getClientEndpoints() {
    return cluster.getClientEndpoints();
  }
}
```

**Comment by lburgazzoli** [comment]
We don't provide support for Junit 4 anymore but only Juni 5: https://github.com/etcd-io/jetcd/blob/master/jetcd-test/src/main/java/io/etcd/jetcd/test/EtcdClusterExtension.java

A PR to fix the documentation would be very appreciated

**Comment by itzg** [comment]
Ah ok. We're not yet using JUnit 4, so I am not equipped to provide a suitable update to the README.

**Comment by dandar3** [comment]
Sorry for hijaking the thread, but it is about Junit5 and using the `EtcdClusterExtension`.

While `jetcd` has moved on from JUnit 4, it appears `testcontainers:1.14.0` hasn't.

`EtcdClusterExtension` will create a cluster through `EtcdClusterFactory.buildCluster()` which uses [`org.testcontainers.containers.Network`](https://github.com/testcontainers/testcontainers-java/blob/1.14.0/core/src/main/java/org/testcontainers/containers/Network.java#L8) that imports `org.junit.rules.TestRule` and if JUnit 4 is banned / removed it will fail at runtime with: `java.lang.NoClassDefFoundError: org/junit/rules/TestRule`

I was trying to remove / ban JUnit 4 altogher from my project. 

I will try anyway to come up with a README.md update for `EtcdClusterExtension` in the Launcher / test section.

**Comment by dandragut** [comment]
Sorry for hijaking the thread, but it is about Junit5 and using the `EtcdClusterExtension`.
While `jetcd` has moved on from JUnit 4, it appears `testcontainers:1.14.0` hasn't.

`EtcdClusterExtension` will create a cluster through `EtcdClusterFactory.buildCluster()` which uses [`org.testcontainers.containers.Network`](https://github.com/testcontainers/testcontainers-java/blob/1.14.0/core/src/main/java/org/testcontainers/containers/Network.java#L8) that imports `org.junit.rules.TestRule` and if JUnit 4 is banned / removed it will fail at runtime with: `java.lang.NoClassDefFoundError: org/junit/rules/TestRule`

I was trying to remove / ban JUnit 4 altogher from my project. 

I will try anyway to come up with a README.md update for `EtcdClusterExtension` in the Launcher / test section.

**Comment by dandragut** [comment]
@lburgazzoli Finally updated `README.md` for Junit 5, please see the PR and let me know if needs any other changes (Y)

**Comment by lburgazzoli** [comment]
Fixed #803
